octoprint_mrbeam/static/js/helpers/working_area_helper.js

```diff
@@ -308,29 +309,49 @@ class WorkingAreaHelper {
                 }`;
             } else {
                 const v = event.target.value;
-                const idxDelimiter = v.search(new RegExp(options.delimiter));
-                if (selStart <= idxDelimiter) {
-                    const v1 = v.substring(0, idxDelimiter);
-                    const newV1 = parseFloat(v1) + val;
-                    event.target.value = `${newV1.toFixed(
-                        options.digits
-                    )}${v.substring(idxDelimiter)}`;
-                } else {
-                    const d = idxDelimiter + options.delimiter.length;
-                    const v2 = v.substring(d);
-                    const newV2 = parseFloat(v2) + val;
-                    event.target.value = `${v.substring(0, d)}${newV2.toFixed(
-                        options.digits
-                    )}`;
+
+                let parts = this._getSubstringAtIndex(
+                    v,
+                    selStart,
+                    options.delimiter
+                );
+                const newV1 = (parseFloat(parts[1]) + val).toFixed(
+                    options.digits
+                );
+                event.target.value = `${parts[0]}${newV1}${parts[2]}`;
+                // maintain cursor position
+                if (selStart > parts[0].length) {
+                    cursorShift = newV1.length - parts[1].length;
                 }
             }
             // restore caret position
-            event.target.selectionStart = selStart;
-            event.target.selectionEnd = selEnd;
+            event.target.selectionStart = selStart + cursorShift;
+            event.target.selectionEnd = selEnd + cursorShift;
             return false; // swallow the default action
         }
         return true;
     }
+
+    static _getSubstringAtIndex(input, index, delimiter = "[^a-zA-Z0-9.,]+") {
+        const iter = input.matchAll(delimiter);
+        let idx1 = 0;
+        let idx2 = input.length;
+        let del;
+        while ((del = iter.next()) !== null) {
+            if (del.done) break;
```

Comment:
I don't care which style a loop is written. Can't see any serious advantage on one of those solutions.

Reply:
I see that it's kind of a thing in javascript apparently
Whenever I see the return value of a variable assignment being used, I start seeing red flags because of how it is missued and dangerous in other older languages.